Rubberduck.Parsing/VBA/DeclarationCaching/DeclarationFinder.cs

```diff
@@ -552,8 +552,10 @@ public ParameterDeclaration FindParameterFromArgument(VBAParser.ArgumentExpressi
             }
 
             var procedure = MatchName(procedureName)
-                .Where(p => AccessibilityCheck.IsAccessible(enclosingProcedure, p))
-                .SingleOrDefault(p => !p.DeclarationType.HasFlag(DeclarationType.Property) || p.DeclarationType.HasFlag(DeclarationType.PropertyGet));
+                .SingleOrDefault(p =>
+                    (!p.DeclarationType.HasFlag(DeclarationType.Property) ||
+                     p.DeclarationType.HasFlag(DeclarationType.PropertyGet)) &&
+                    p.References.Any(r => callStmt == r.Context.GetAncestor<VBAParser.CallStmtContext>()));
```

Comment:
The idea was to only get the functions and property getters, since that's what gets invoked in a call statement. Reading the query again, it's indeed weirdly formulated. The "not property" part does make it work for functions though... in a non-obvious way that should probably just explicitly filter for `Function` and `PropertyGet`.

Reply:
For some reason I was thinking that I didn't have the QMN I needed to build a qualified selection there, but I missed the `enclosingProcedure` parameter. Rewritten to use the qualified selection. This also means that the `DeclarationType` filtering is moot because it's implicit in the identifier reference.

Reply:
I think my initial point was not really understood. Given that the method is now on the `DeclarationFinder` and called `FindParameterFromArgument` I would expect it to work for all call statements and all non-traditional index expressions.